src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -2080,16 +2144,475 @@ export class SignAccountOpController extends EventEmitter implements ISignAccoun
+   *   second one is the signature of the transaction that will be executed
+   *   by the smart account.
+   *   3. For smart accounts that broadcast the ERC-4337 way.
+   *   4. for smart accounts, when the Relayer does the broadcast.
+   *
+   */
+  async #broadcast() {
+    if (this.status?.type !== SigningStatus.Done) {
+      this.throwBroadcastAccountOp({
+        message: 'Pending broadcast. Please try again in a bit.'
+      })
+      return
+    }
+    const accountOp = this.accountOp
+    const estimation = this.estimation.estimation
+    const actionId = this.fromActionId
+    const bundlerSwitcher = this.bundlerSwitcher
+    const contactSupportPrompt = 'Please try again or contact support if the problem persists.'
+
+    if (
+      !accountOp ||
+      !estimation ||
+      !actionId ||
+      !accountOp.signingKeyAddr ||
+      !accountOp.signingKeyType ||
+      !accountOp.signature ||
+      !bundlerSwitcher ||
+      !accountOp.gasFeePayment
+    ) {
+      const message = `Missing mandatory transaction details. ${contactSupportPrompt}`
+      return this.throwBroadcastAccountOp({ message })
+    }
+
+    const account = this.#accounts.accounts.find((acc) => acc.addr === accountOp.accountAddr)
+
+    if (!this.provider) {
+      const networkName = this.#network?.name || `network with id ${accountOp.chainId}`
+      const message = `Provider for ${networkName} not found. ${contactSupportPrompt}`
+      return this.throwBroadcastAccountOp({ message })
+    }
+
+    if (!account) {
+      const addr = shortenAddress(accountOp.accountAddr, 13)
+      const message = `Account with address ${addr} not found. ${contactSupportPrompt}`
+      return this.throwBroadcastAccountOp({ message })
+    }
+
+    if (!this.#network) {
+      const message = `Network with id ${accountOp.chainId} not found. ${contactSupportPrompt}`
+      return this.throwBroadcastAccountOp({ message })
+    }
+
+    this.broadcastStatus = 'LOADING'
+    await this.forceEmitUpdate()
+
+    const accountState = await this.#accounts.getOrFetchAccountOnChainState(
+      accountOp.accountAddr,
+      accountOp.chainId
+    )
+    const baseAcc = getBaseAccount(
+      account,
+      accountState,
+      this.#keystore.getAccountKeys(account),
+      this.#network
+    )
+    let transactionRes: {
+      txnId?: string
+      nonce: number
+      identifiedBy: AccountOpIdentifiedBy
+    } | null = null
+
+    // broadcasting by EOA is quite the same:
+    // 1) build a rawTxn 2) sign 3) broadcast
+    // we have one handle, just a diff rawTxn for each case
+    const rawTxnBroadcast = [
+      BROADCAST_OPTIONS.bySelf,
+      BROADCAST_OPTIONS.bySelf7702,
+      BROADCAST_OPTIONS.byOtherEOA,
+      BROADCAST_OPTIONS.delegation
+    ]
+
+    if (rawTxnBroadcast.includes(accountOp.gasFeePayment.broadcastOption)) {
+      const multipleTxnsBroadcastRes = []
+      const senderAddr = BROADCAST_OPTIONS.byOtherEOA
```

Comment:
This line has a bug that I fixed in the gas tank PR but if this one gets merged ahead of the gas tank, it will disappear. So let's add it:
```
const senderAddr = accountOp.gasFeePayment.broadcastOption === BROADCAST_OPTIONS.byOtherEOA
```